swagger/fs_wrapper.go

```diff
@@ -0,0 +1,95 @@
+package swagger
+
+import (
+	"bytes"
+	"html/template"
+	"io"
+	"io/fs"
+	"strings"
+	"time"
+)
+
+type fsWrapper struct {
+	fs           fs.FS
+	jsonFileName string
+}
+
+func FsWrapper(fs fs.FS, jsonFilePath string) *fsWrapper {
+	return &fsWrapper{
+		fs:           fs,
+		jsonFileName: jsonFilePath,
+	}
+}
+
+func (f *fsWrapper) Open(name string) (fs.File, error) {
+	if strings.HasSuffix(name, "index.html") {
+		tpl, err := template.ParseFS(f.fs, "index.gohtml")
+		if err != nil {
+			return nil, err
+		}
+		data := map[string]interface{}{
+			"openApiJson": strings.TrimPrefix(f.jsonFileName, "/"),
```

Comment:
See related comment.
Instead of trimming the slash prefix to add it later in the html file, we should ensute that it is present or else add it now.